File: src/Java/1-100/98.ValidateBinarySearchTree.java

import java.util.List;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.Comparator;
import java.util.HashMap;
import java.util.Queue;
import java.util.HashSet;
import java.util.LinkedList;


public class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode(int x) { val = x; }
}

class Solution {
    long tmp = Long.MIN_VALUE;
    public boolean isValidBST(TreeNode root) {
        if(root == null) return true;
        if (isValidBST(root.left)) {
            if (tmp < root.val) {
                tmp = root.val;
                return isValidBST(root.right);
            }
        }
        return false;
   }
}

File: src/CSharp/1-100/98.ValidateBinarySearchTree.cs

using System;
using System.Linq;
using System.Text;
using System.Collections.Generic;


namespace PeefyLeetCode.ValidateBinarySearchTree
{

public class TreeNode {
    public int val;
    public TreeNode left;
    public TreeNode right;
    public TreeNode(int x) { val = x; }
}

public class Solution {
    long tmp = long.MinValue;
    public bool IsValidBST(TreeNode root) {
                if(root == null) return true;
        if (IsValidBST(root.left)) {
            if (tmp < root.val) {
                tmp = root.val;
                return IsValidBST(root.right);
            }
        }
        return false;
    }
}
}